### PR TITLE
feat: expose deployment to serve Executors

### PR DESCRIPTION
**Pull request by alaeddine-13**
This PR makes it possible to serve Executors using Deployment

- [x] Serve Executors with deployment
- [x] add overload for Deployment
- [x] Unify behavior with Flow ?
    * print info
    * reload
    * refactor common code into `BaseOrchestrator` class
- [x] Tests: single Executor deployment (Executor python class, docker image, yaml), test for reload (if reload is implemented), dynamic batching (like in https://github.com/jina-ai/jina/blob/1fab6208a576b835f466fc7c7cac926b1b1cf185/tests/unit/serve/executors/test_executor.py#L49) ? in tests/unit/orchestrate/deployments/test_deployments.py or tests/integration/deployments/test_deployment.py
- [x] Expose Deployment class
- [x] Documentation

### CODE REVIEW

File: jina/orchestrate/deployments/__init__.py [hunk 1]
```diff
@@ -1,158 +1,49 @@
...
 import copy
 import json
+import multiprocessing
 import os
 import re
 import subprocess
-import asyncio
-
-from abc import abstractmethod
+import threading
```
Comment (JoanFM) [inline jina/orchestrate/deployments/__init__.py:8]: also this

File: jina/orchestrate/deployments/__init__.py [hunk 2]
```diff
@@ -1065,3 +1274,142 @@ def _mermaid_str(self) -> List[str]:
...
+
+        all_panels.append(
+            Panel(
+                address_table,
+                title=':link: [b]Endpoint[/]',
+                expand=False,
+            )
+        )
+
+        if self.protocol == GatewayProtocolType.HTTP:
```
Comment (JoanFM) [inline jina/orchestrate/deployments/__init__.py:1374]: please remove protocol for now

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/orchestrate/deployments/__init__.py (hunk 1 above)
Comment (alaeddine-13) [reply jina/orchestrate/deployments/__init__.py:7]: not really, it's used somewhere else

File: docs/concepts/executor/serve.md [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
 # Serve
 
-{class}`~jina.Executor`s can be served - and remotely accessed - directly, without instantiating a Flow manually.
-This is especially useful when debugging an Executor in a remote setting. It can also be used to run external/shared Executors to be used in multiple Flows.
+{class}`~jina.Executor`s can be served and accessed over gRPC, allowing you to use them to create a gRPC-based service for various tasks such as model inference, data processing, generative AI, and search services.
 
 There are different options for deploying and running a standalone Executor:
-* Run the Executor directly from Python with the `.serve()` class method
+* Run the Executor directly from Python with the {class}`~jina.orchestrate.deployments.Deployment` class
 * Run the static {meth}`~jina.serve.executors.BaseExecutor.to_kubernetes_yaml()` method to generate K8s deployment configuration files
```
Comment (alexcg1) [inline docs/concepts/executor/serve.md:8]:
```suggestion
* Run the static {meth}`~jina.serve.executors.BaseExecutor.to_kubernetes_yaml()` method to generate Kubernetes deployment configuration files
```

File: docs/concepts/executor/serve.md [hunk 4]
```diff
@@ -28,27 +34,85 @@ Both served and shared Executors can be used as part of a Flow, by adding them a
 ````
 
 ## Serve directly
-An {class}`~jina.Executor` can be served using the {meth}`~jina.serve.executors.BaseExecutor.serve` method:
+An {class}`~jina.Executor` can be served using the {class}`~jina.orchestrate.deployments.Deployment` class.
+
+The {class}`~jina.orchestrate.deployments.Deployment` class aims at separating the deployment configuration from the 
+serving logic.
```
Comment (alexcg1) [inline docs/concepts/executor/serve.md:40]:
```suggestion
The {class}`~jina.orchestrate.deployments.Deployment` class aims to separate the deployment configuration from the serving logic.

File: docs/concepts/executor/serve.md [hunk 5]
```diff
@@ -28,27 +34,85 @@ Both served and shared Executors can be used as part of a Flow, by adding them a
...
 
 ## Serve directly
-An {class}`~jina.Executor` can be served using the {meth}`~jina.serve.executors.BaseExecutor.serve` method:
+An {class}`~jina.Executor` can be served using the {class}`~jina.orchestrate.deployments.Deployment` class.
+
+The {class}`~jina.orchestrate.deployments.Deployment` class aims at separating the deployment configuration from the 
+serving logic. 
+In other words:
+* the Executor cares about defining the logic to serve, which endpoints to define and what data to accept.
+* the Deployment layer cares about how to orchestrate this service, how many replicas or shards,...
```
Comment (alexcg1) [inline docs/concepts/executor/serve.md:43]:
```suggestion
* the Deployment layer cares about how to orchestrate this service, how many replicas or shards, etc.
```

File: docs/concepts/executor/serve.md [hunk 6]
```diff
@@ -28,27 +34,85 @@ Both served and shared Executors can be used as part of a Flow, by adding them a
...
+
+This separation also aims at enhancing the reusability of Executors: the same implementation of an Executor can be
```
Comment (alexcg1) [inline docs/concepts/executor/serve.md:45]:
```suggestion
This separation also aims to enhance the reusability of Executors: the same implementation of an Executor can be 
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/concepts/executor/serve.md [hunk 7]
```diff
@@ -129,99 +191,3 @@ The Executor you use needs to be already containerized and stored in an accessib
 ````
 
 (reload-executor)=
-## Reload Executor
```
Comment (JoanFM) [inline docs/concepts/executor/serve.md:132]: why is this removed?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/concepts/executor/serve.md (hunk 7 above)
Comment (alaeddine-13) [reply docs/concepts/executor/serve.md:132]: content duplicated here https://feat-deployment-serve--jina-docs.netlify.app/concepts/executor/hot-reload/

File: docs/concepts/executor/serve.md [hunk 8]
```diff
@@ -129,99 +190,3 @@ The Executor you use needs to be already containerized and stored in an accessib
 ````
 
 (reload-executor)=
```
Comment (alaeddine-13) [inline docs/concepts/executor/serve.md:192]:
```suggestion
```

File: docs/concepts/executor/serve.md (hunk 7 above)
Comment (alaeddine-13) [reply docs/concepts/executor/serve.md:132]: removed

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

**Review by JoanFM — DISMISSED** [review]

**Review by JoanFM — APPROVED** [review]
